subsys/bluetooth/mesh/crypto_mtls.c

```diff
@@ -357,3 +297,145 @@ __weak psa_status_t mbedtls_psa_external_get_random(mbedtls_psa_external_random_
+{
+	psa_key_attributes_t key_attributes = PSA_KEY_ATTRIBUTES_INIT;
+	psa_status_t status;
+	int err = 0;
+
+	switch (type) {
+	case BT_MESH_KEY_TYPE_ECB:
+		psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_VOLATILE);
+		psa_set_key_usage_flags(&key_attributes,
+			PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_DECRYPT);
+		psa_set_key_algorithm(&key_attributes, PSA_ALG_ECB_NO_PADDING);
+		break;
+	case BT_MESH_KEY_TYPE_CCM:
+		psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_VOLATILE);
+		psa_set_key_usage_flags(&key_attributes,
+			PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_DECRYPT);
+		psa_set_key_algorithm(&key_attributes,
+			PSA_ALG_AEAD_WITH_AT_LEAST_THIS_LENGTH_TAG(PSA_ALG_CCM, 4));
+		break;
+	case BT_MESH_KEY_TYPE_CMAC:
+		psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_VOLATILE);
+		psa_set_key_usage_flags(&key_attributes, PSA_KEY_USAGE_SIGN_MESSAGE);
+		psa_set_key_algorithm(&key_attributes, PSA_ALG_CMAC);
+		break;
+	case BT_MESH_KEY_TYPE_NET:
+		if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+			psa_key_id_t key_id = bt_mesh_user_keyid_alloc();
+
+			if (key_id == PSA_KEY_ID_NULL) {
+				return -ENOMEM;
+			}
+
+			psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_PERSISTENT);
+			psa_set_key_id(&key_attributes, key_id);
+		} else {
+			psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_VOLATILE);
+		}
+		psa_set_key_usage_flags(&key_attributes, PSA_KEY_USAGE_EXPORT);
+		break;
+	case BT_MESH_KEY_TYPE_APP:
+	case BT_MESH_KEY_TYPE_DEV:
+		if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+			psa_key_id_t key_id = bt_mesh_user_keyid_alloc();
+
+			if (key_id == PSA_KEY_ID_NULL) {
+				return -ENOMEM;
+			}
+
+			psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_PERSISTENT);
+			psa_set_key_id(&key_attributes, key_id);
+		} else {
+			psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_VOLATILE);
+		}
+		psa_set_key_usage_flags(&key_attributes,
+			PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_DECRYPT | PSA_KEY_USAGE_EXPORT);
+		psa_set_key_algorithm(&key_attributes,
+			PSA_ALG_AEAD_WITH_AT_LEAST_THIS_LENGTH_TAG(PSA_ALG_CCM, 4));
+		break;
+	default:
+		return -EIO;
+	}
+
+	psa_set_key_type(&key_attributes, PSA_KEY_TYPE_AES);
+	psa_set_key_bits(&key_attributes, 128);
+
+	status = psa_import_key(&key_attributes, in, 16, &out->key);
+	err = status == PSA_SUCCESS ? 0 :
+		status == PSA_ERROR_ALREADY_EXISTS ? -EALREADY : -EIO;
+	psa_reset_key_attributes(&key_attributes);
+
+	return err;
+}
+
+int bt_mesh_key_export(uint8_t out[16], const struct bt_mesh_key *in)
+{
+	size_t data_length;
+
+	if (psa_export_key(in->key, out, 16, &data_length) != PSA_SUCCESS) {
+		return -EIO;
+	}
+
+	if (data_length != 16) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int bt_mesh_key_destroy(struct bt_mesh_key *key)
+{
+	int err = 0;
+
+	if (psa_destroy_key(key->key) != PSA_SUCCESS) {
+		return -EIO;
+	}
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		err = bt_mesh_user_keyid_free(key->key);
```

Comment:
```suggestion
		return bt_mesh_user_keyid_free(key->key);
```
And remove int err = 0; and return 0 at the end